### PR TITLE
fix(Table): First level header is no longer required

### LINKED ISSUE #153: Table: columns first level Header is not respected.
In order to get headers to be displayed in the table, we must actually create a root group (which is never displayed)
It is a bit unexpected behavior.

#### Environment
- __Package version(s)__: 1.7.2
- __Operating System__: Windows 10
- __Browser name and version__: Chrome, 91.0.4472.114

#### Code Sandbox
Link to a minimal repro: https://codesandbox.io/s/shy-wind-1sp5q?file=/src/App.js
This pits react-table example against itwinUi table, click on the Toggle Header Level to add/remove the grouping, the same data is fed to their example and iui table, notice that, the groupings are off.

#### Steps to reproduce
Please see the repo, compare the columns definition between grouped and not grouped and the behavior.

#### Actual behavior
In the sandbox, I get either "Ungrouped" headers, or no headers.

#### Expected behavior
In the sandbox, I expect a grouping level, and an "Ungrouped" header.

#### Additional information
There is a `.splice(1)` in the Table header code that seems to cause this issue, not sure why it is there.

### PR DESCRIPTION
**Pull request by adamhock**
Continuation from #911 

Fixes #153

### CODE REVIEW

File: packages/iTwinUI-react/src/core/Table/Table.tsx [hunk 1]
```diff
@@ -828,23 +822,30 @@ export const Table = <
...
+            (header) =>
+              !header.id.includes('iui-table-checkbox-selector_placeholder') &&
+              !header.id.includes('iui-table-expander_placeholder'),
+          );
+          const headerGroupProps = headerGroup.getHeaderGroupProps({
+            className: 'iui-table-row',
+          });
+          return (
+            <div
+              className='iui-table-header-wrapper'
```
Comment (mayank99) [inline packages/iTwinUI-react/src/core/Table/Table.tsx:837]: Shouldn't this `header-wrapper` wrap all groups?

**Review by mayank99 — COMMENTED** [review]
PR title should be renamed if this is not adding full support for grouped headers.

File: packages/iTwinUI-react/src/core/Table/Table.tsx (hunk 1 above)
Comment (adamhock) [reply packages/iTwinUI-react/src/core/Table/Table.tsx:837]: This is what Horiztonal Scroll looks like when the `Header` is specified now
image

After wrapping all of the header groups in `iui-table-header-wrapper`, the result looks like this
TableRowHeaderHorizontalScroll

File: packages/iTwinUI-react/src/core/Table/Table.test.tsx [hunk 2]
```diff
@@ -3932,21 +3655,16 @@ it('should add expander column manually', () => {
...
+      Header: 'Name',
+      accessor: 'name',
+    },
+    {
+      id: 'description',
+      Header: 'Description',
+      accessor: 'description',
     },
   ];
   const { container } = renderComponent({
```
Comment (elephantcatdog) [inline packages/iTwinUI-react/src/core/Table/Table.test.tsx:3670]: Lines 3698 and 3732 need to be updated to remove the `Header: 'Table'` thing

File: packages/iTwinUI-react/src/core/Table/Table.tsx (hunk 1 above)
Comment (mayank99) [reply packages/iTwinUI-react/src/core/Table/Table.tsx:837]: i think this is a css bug. the wrapper needs flex-direction: column probably

File: packages/iTwinUI-react/src/core/Table/Table.test.tsx (hunk 2 above)
Comment (adamhock) [reply packages/iTwinUI-react/src/core/Table/Table.test.tsx:3670]: Removed

File: apps/storybook/src/Table.stories.tsx [hunk 3]
```diff
@@ -3130,80 +2983,73 @@ export const ColumnManager: Story<Partial<TableProps>> = (args) => {
...
+            <>{props.row.original.startDate.toLocaleDateString('en-US')}</>
+          );
+        },
+      },
+      {
+        id: 'endDate',
+        Header: 'End date',
+        accessor: 'endDate',
+        Cell: (props: CellProps<TableStoryDataType>) => {
+          return <>{props.row.original.endDate.toLocaleDateString('en-US')}</>;
```
Comment (elephantcatdog) [inline apps/storybook/src/Table.stories.tsx:3022]: This return statement no longer has () around it. Is that okay?

Comment (adamhock) [reply apps/storybook/src/Table.stories.tsx:3022]: I believe this is some `prettier` print issue

Comment (elephantcatdog) [reply apps/storybook/src/Table.stories.tsx:3022]: ah ok

**Comment by elephantcatdog** [comment]
Seems ok to me. I approve whenever Mayank does

File: packages/iTwinUI-react/src/core/Table/Table.tsx (hunk 1 above)
Comment (elephantcatdog) [reply packages/iTwinUI-react/src/core/Table/Table.tsx:837]: @adamhock Does adding `flex-direction` fix the problem with `header-wrapper` wrapping all groups?

Comment (bentleyvk) [reply packages/iTwinUI-react/src/core/Table/Table.tsx:837]: I don't think we are supporting columns grouping so there is no need to fix this.

Comment (adamhock) [reply packages/iTwinUI-react/src/core/Table/Table.tsx:837]: Adding `flex-direction` does solve this problem, but there's another problem that occurs after this change is made.

TableRowHeaderHorizontalScroll

Comment (mayank99) [reply packages/iTwinUI-react/src/core/Table/Table.tsx:837]: > I don't think we are supporting columns grouping so there is no need to fix this.

Ok, so lets merge this as-is. But we need to clear in our messaging that we do not support grouped headers yet.

**Review by mayank99 — APPROVED** [review]

**Pull request merged**